src/RedisOutputCachingMiddleware/RedisOutputCachingMiddleware.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.VisualBasic;
+using Microsoft.Web.Redis;
+using StackExchange.Redis;
+
+namespace RedisOutputCachingMiddleware
+{
+    public class OutputCachingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IDatabase _cache;
+        private int _ttl;
+
+        // optional eviction time, default to 1 day if not defined 
+        public OutputCachingMiddleware(RequestDelegate next, string redisConnectionString, int ttl = 86400)
```

Comment:
For clarity,  you can default the ttl param as: 
`int ttl = TimeSpan.FromDays(1).TotalSeconds`